lib/assert.js

```diff
@@ -388,22 +498,14 @@ function compareBranch(
 ) {
   // Check for Map object equality
   if (isMap(actual) && isMap(expected)) {
-    if (actual.size !== expected.size) {
-      return false;
-    }
-    const safeIterator = FunctionPrototypeCall(SafeMap.prototype[SymbolIterator], actual);
-
-    comparedObjects ??= new SafeWeakSet();
+    return compareMaps(actual, expected, comparedObjects);
+  }
 
-    for (const { 0: key, 1: val } of safeIterator) {
-      if (!MapPrototypeHas(expected, key)) {
-        return false;
-      }
-      if (!compareBranch(val, MapPrototypeGet(expected, key), comparedObjects)) {
-        return false;
-      }
-    }
-    return true;
+  if (
+    (ArrayBufferIsView(actual) && ArrayBufferIsView(expected)) ||
+    (actual instanceof ArrayBuffer && expected instanceof ArrayBuffer)
```

Comment:
Let's not use `instanceof`, it doesn't work with values coming from other realms. Instead, let's use `isArrayBuffer` from `internal/util/types`